clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -1635,8 +1656,13 @@ bool tools::addSanitizerRuntimes(const ToolChain &TC, const ArgList &Args,
   }
   // If there is a static runtime with no dynamic list, force all the symbols
   // to be dynamic to be sure we export sanitizer interface functions.
-  if (AddExportDynamic)
-    CmdArgs.push_back("--export-dynamic");
+  if (AddExportDynamic) {
+    if (!TC.getTriple().isOSAIX())
+      CmdArgs.push_back("--export-dynamic");
+    else
+      llvm::report_fatal_error("Sanitizer interface functions must be exported "
```

Comment:
clang/lib/Driver has proper error reporting mechanism (D.Diag). Should not use `llvm::report_fatal_error`. I have noticed that only IBM OS code (z/OS, AIX)  uses `llvm::report_fatal_error`, which should hopefully be fixed.

Reply:
This error text is entirely confusing anyway. What is a user to do about it?
IIUC, this error arises when the dynamic lists are not found (e.g., because of installation/configuration errors).
If that is the case, then a better error message would be:
> Cannot link "&lt;sanitizer>": Export file missing from resource directories."